validator/src/main/java/com/google/daq/mqtt/registrar/LocalDevice.java

```diff
@@ -244,19 +263,41 @@ public void validateExpected() {
     exceptionMap.throwIfNotEmpty();
   }
 
+  private void deepMerge(Map<String, Object> map1, Map<String, Object> map2) {
```

Comment:
nit -- rename parameters to indicate which is target vs. src -- the IDE can be good about hinting based on names, so it does provide useful context for somebody using the function